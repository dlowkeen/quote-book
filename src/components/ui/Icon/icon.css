.icon {
  display: inline-block;
  vertical-align: middle;

  path {
    fill: currentColor;
  }
}
