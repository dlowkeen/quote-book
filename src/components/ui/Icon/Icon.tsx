import React from 'react';
import ReactSVG from 'react-svg';

import { IMAGES_DIR } from '../../../config';

// Import hashed filenames of image assets to use them as 'src' attribute for images
// This is made possible by the module 'babel-plugin-import-static-files'
import imagePaths from '../../../public';
// const imagePaths: any = images;

import * as styles from './icon.css';

// Creates an inline-svg icon.
// Prop "type" should be the name of the icon file (minus extension) -- so <Icon type='check' /> will get you '/check-[hash].svg'

interface IIconProps {
  type: string;
}

const Icon: React.FunctionComponent<IIconProps> = ({ type }) => {
  return (
    <div className={styles.icon}>
      <ReactSVG src={`${IMAGES_DIR}${imagePaths[type]}`} />
    </div>
  );
};

export default Icon;
