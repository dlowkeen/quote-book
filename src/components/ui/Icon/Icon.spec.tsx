import React from 'react';
import renderer from 'react-test-renderer';

import Icon from './Icon';

describe('The Icon component', () => {
  it('renders as expected', () => {
    const tree = renderer.create(<Icon type='iconType' />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
