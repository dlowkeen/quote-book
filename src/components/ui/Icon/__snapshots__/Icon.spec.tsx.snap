// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Icon component renders as expected 1`] = `
<div
  className="icon__icon__3WuBE"
>
  <div />
</div>
`;
